### PR TITLE
Feature/s05 fullcalendar front

### PR DESCRIPTION
**Pull request by Carcali**
📅 Ajout de FullCalendar en front :

- mises à jours en fonction des reviews (suppression/modification des commentaires, merge develop, remise en place du css dans DashLayout...)
- un fichier séparé pour le planning des trainer (PlanningTrainer) et des owner (PlanningOwner)
- un css commun pour le moment (Planning.scss)
- la mise à jour du style du planning en le passant du bleu au beige
- isolation de la function checkIsMobile dans un hook disponible

### CODE REVIEW

File: client/src/main.tsx [hunk 1]
```diff
@@ -76,7 +77,7 @@ const router = createBrowserRouter([
 						children: [
 							{
 								index: true,
-								element: <Navigate to="planning" replace />,
+								element: <PlanningTrainer />,
```
Comment (FlorenceBuchelet) [inline client/src/main.tsx:80]: Ca devrait être sur owner/planning et non juste owner (et il le faut aussi pour trainer)

File: client/src/pages/Planning/PlanningTrainer.scss [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        color: $primary-light !important;
+        font-weight: $regular;
+
+        &:hover {
+            background-color: $blue-500 !important;
+            transition: all ease-in-out 0.3s;
+        }
+    }
+
+    /* Configuration de la ligne right */
```
Comment (FlorenceBuchelet) [inline client/src/pages/Planning/PlanningTrainer.scss:264]: joli franglais ahah, restons dans une seule langue, ok ? 😂
D'ailleurs il me semble que nous avions dit tout en anglais pour le projet, du coup est-ce qu'il ne faudrait pas passer tes commentaires en anglais ?

File: client/src/pages/Planning/PlanningTrainer.scss [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    .fc .fc-button {
+        padding: 0.8rem !important;
+    }
+
+    .fc .fc-today-button {
+        padding: 0.8rem !important;
+        margin: 0 !important;
+    }
+
+    /* Nouvelles modifications pour le calendrier responsive */
```
Comment (FlorenceBuchelet) [inline client/src/pages/Planning/PlanningTrainer.scss:306]: /* Ajustements supplémentaires pour la cohérence mobile */ et/* Nouvelles modifications pour le calendrier responsive */ ne sont pas des commentaires très clairs sur ce qu'ils vont contenir "nouvelles modifs" ça ne veut pas dire grand chose, quand le projet aura plusieurs années, il sera toujours écrit "nouveau" mais ça n'aura plus de sens, mieux vaut préciser de quoi il s'agit exactement.

File: client/src/pages/Planning/PlanningTrainer.scss
Comment (FlorenceBuchelet) [inline client/src/pages/Planning/PlanningTrainer.scss:1]: Super le responsive !
Par contre, pourquoi est-ce que la couleur change entre le mobile et le desktop ? J'aime mieux la version plus claire du mobile perso (mais le plus important à mes yeux c'est que les deux soit cohérents).
Et ça fait beaucoup de `!important`, c'est pour override le style par défaut de fullcalendar ?

File: client/src/pages/Planning/PlanningTrainer.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import timeGridPlugin from "@fullcalendar/timegrid";
+import listPlugin from "@fullcalendar/list";
+import frLocale from "@fullcalendar/core/locales/fr";
+import interactionPlugin from "@fullcalendar/interaction";
+
+function PlanningTrainer() {
+	const [currentView] = useState("dayGridMonth");
+	const [isMobile, setIsMobile] = useState(false);
+
+	// Fonction pour vérifier si l'écran est en mode mobile
```
Comment (FlorenceBuchelet) [inline client/src/pages/Planning/PlanningTrainer.tsx:17]: A nouveau, les commentaires faut qu'on en parle.

File: client/src/pages/Planning/PlanningTrainer.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+		checkIsMobile();
+
+		// Ajouter l'écouteur d'événement
+		window.addEventListener("resize", checkIsMobile);
+
+		// Nettoyer l'écouteur lors du démontage du composant
+		return () => {
+			window.removeEventListener("resize", checkIsMobile);
+		};
+	}, []);
```
Comment (FlorenceBuchelet) [inline client/src/pages/Planning/PlanningTrainer.tsx:33]: Ce serait intéressant d'isoler la logique d'un check de format de l'écran pour qu'il puisse être réutilisé ailleurs !

File: client/src/pages/Planning/PlanningTrainer.tsx
Comment (FlorenceBuchelet) [inline client/src/pages/Planning/PlanningTrainer.tsx:1]: Absolument fou à quel point le code pour fullcalendar est cours comparé à ce qu'il fait, chouette lib !

File: client/src/layouts/Dashboard/DashLayout.scss [hunk 6]
```diff
@@ -21,6 +21,8 @@ $dashInlinePadding: $minPaddingInline;
     flex-direction: row;
     justify-content: space-between;
     align-items: center;
+    padding-block: 2%;
```
[the remark this answers is not in the document]
Comment (Carcali) [reply client/src/layouts/Dashboard/DashLayout.scss:24]: ✅

File: client/src/layouts/Dashboard/DashLayout.scss [hunk 7]
```diff
@@ -80,9 +82,9 @@ $dashInlinePadding: $minPaddingInline;
     flex-direction: column;
     justify-content: space-between;
     align-items: center;
-    padding-inline: $dashInlinePadding;
-    padding-block: 1vh;
-
+    padding: $dashInlinePadding;
```
[the remark this answers is not in the document]
Comment (Carcali) [reply client/src/layouts/Dashboard/DashLayout.scss:85]: ✅ 
Les deux paddings d'origine ont été remis :
    padding-inline: $dashInlinePadding;
    padding-block: 1vh;

File: client/src/main.tsx (hunk 1 above)
Comment (Carcali) [reply client/src/main.tsx:89]: ✅

File: client/src/pages/Planning/PlanningTrainer.scss (hunk 3 above)
Comment (Carcali) [reply client/src/pages/Planning/PlanningTrainer.scss:306]: ✅

File: client/src/pages/Planning/PlanningTrainer.tsx (hunk 4 above)
Comment (Carcali) [reply client/src/pages/Planning/PlanningTrainer.tsx:17]: ✅

File: client/src/components/_atoms/Button/Button.tsx [hunk 8]
```diff
@@ -58,9 +58,7 @@ export default function Button({
 			className={`button ${buttonClassName} ${className}`}
 			onClick={onClick}
 		>
-			{type === "invite" && !children
-				? "+ Inviter un client à s'inscrire"
-				: children}
+			{type === "invite" && !children ? "+ Ajouter un événement" : children}
```
[the remark this answers is not in the document]
Comment (Carcali) [reply client/src/components/_atoms/Button/Button.tsx:61]: ✅

File: client/src/pages/Planning/PlanningTrainer.tsx (hunk 5 above)
Comment (Carcali) [reply client/src/pages/Planning/PlanningTrainer.tsx:33]: ✅ 
Création du fichier checkIsMobile.ts et utilisation du hook dans le composant Planning pour le moment

**Review by FlorenceBuchelet — APPROVED** [review]
